yapapi/runner/events.py

```diff
@@ -0,0 +1,117 @@
+"""Code for representing and logging events in Golem computation."""
+import asyncio
+from dataclasses import dataclass, field
+from enum import Enum, auto
+from typing import Any, ClassVar, Mapping, Optional, TypeVar, Union
+
+from typing_extensions import Protocol
+
+
+ResourceId = Union[int, str]
+EventType = str
+
+
+class ResourceType(Enum):
+    """Resource types for events."""
+
+    SUBSCRIPTION = auto()
+    PROPOSAL = auto()
+    AGREEMENT = auto()
+    ACTIVITY = auto()
+    WORKER = auto()
+    TASK = auto()
+
+
+@dataclass(frozen=True)
+class _Event:
+    """Represents an event in a Golem computation."""
+
+    resource_type: ResourceType
+    event_type: EventType
+    resource_id: Optional[ResourceId] = None
+    info: Mapping[str, Any] = field(default_factory=dict)
+
+    event_type_to_readable: ClassVar[Mapping[str, str]] = {
+        "recv": "received",
```

Comment:
shouldn't the keys here also be an enum?

Reply:
I'm not sure what were the intentions of the original author. The keys are not exhaustive now, e.g. there are calls to `emit_progress()` with values not in this dictionary, so I had an impression that this is an "open" type that can be extended, so perhaps an enum would not be appropriate?

Anyway, I considered making this an enum but also defining string constants, to make sure the messages are uniform:
```
class EventType:
   RECV = "received"
   ... etc. ...
```
but couldn't see much value in this as each message is at the moment used only once. In the end, I couldn't decide how to change this so I left is without changes.

Reply:
Now I'm mostly in favor of removing this dict altogether and just using full message text in calls to `emit_progress()`, e.g. `emit_progress(res_type, "failed to subscribe to market events")` instead of `emit_progress(res_type, "fail-subscribe")`.

What do you think?

Reply:
maybe just let's leave it in the form that you have here and see in the future how it's going to be used - maybe it'll change the balance betewen enum or plain strings and it'll be easier to decide

Reply:
So in the end I decided to use just a single enum instead of (resource_type, event_type) pair of strings (see the PR's description).